contact_map/contact_count.py

```diff
@@ -57,6 +57,49 @@ def _patch_from_spmatrix(cls, data):  # -no-cov-
     pd.core.arrays.SparseArray.from_spmatrix = classmethod(_patch_from_spmatrix)
 # TODO: this is the end of what to remove when pandas is fixed
 
+
+def _int_or_range_to_tuple(posible_int):
+    if isinstance(posible_int, collections.abc.Iterable):
+        return (posible_int[0], posible_int[1])
+    else:
+        return (0, posible_int)
+
+
+def _get_total_counter_range(counter):
+    numbers = sorted([i for key in counter.keys() for i in key])
+    if len(numbers) == 0:
+        return (0, 0)
+    return (numbers[0], numbers[-1]+1)
+
+
+def _get_low_high_counter_range(counter):
+    """Give the (min, max + 1) for both the low and high keys in counter"""
+    keys = [tuple(sorted(list(i))) for i in counter.keys()]
+    if len(keys) == 0:
+        return (0, 0), (0, 0)
+
+    lows = sorted([i for i, j in keys])
+    highs = sorted([j for i, j in keys])
+    return (lows[0], lows[-1]+1), (highs[0], highs[-1]+1)
+
+
+def _get_sorted_counter_range(counter):
+    """Return smallest range, longest range for the low and high counter"""
+    low, high = _get_low_high_counter_range(counter)
+    if low[1]-low[0] > high[-1]-high[0]:
+        return high, low
+    return low, high
```

Comment:
Readability here would be improved with explicit `else` statement and single return. That is, something like:

```python
if ...:
    n_x, n_y = high, low
else:
    n_x, n_y = low, high
return n_x, n_y
```